### PR TITLE
Chore: Updated logging / constant strings / variable names in Apply Controller.

### PR DESCRIPTION
**Pull request by Ealianis**
Quality control on log standards:

- Uses KObjRef for structured logging 
- Standardizing string messages as much as possible. Generic messages where context (resource type) is available via some other field.
- Logs out Namespace + name of resources (including Work resources)
  - This is necessary as the namespace of Work will not be "default" in prod.
- Renamed local variables to match usage e.g. ManifestObj in lieu of WorkObj. So as not to confuse Work resource.

Fixes #
I have:
- [x] Read and followed Caravel's [Code of conduct](https://github.com/Azure/k8s-work-api/blob/master/code-of-conduct.md).
- [x] Run `make reviewable` to ensure this PR is ready for review.

### CODE REVIEW

File: pkg/controllers/apply_controller.go [hunk 1]
```diff
@@ -46,15 +46,26 @@ import (
...
+	messageResourceCreateSucceeded   = "resource create succeeded"
+	messageResourceJSONMarshalFailed = "resource JSON marshaling failed"
+	messageResourceNotOwnedByWorkAPI = "resource not owned by Work-API"
+	messageResourcePatchFailed       = "resource patch failed"
+	messageResourcePatchSucceeded    = "resource patch succeeded"
+	messageResourceRetrieveFailed    = "resource retrieval failed"
+	messageResourceStateInvalid      = "resource state is invalid"
+	messageResourceSpecModified      = "resource spec modified"
+	messageResourceUpdateFailed      = "resource update failed"
+	messageResourceUpdateSucceeded   = "resource update succeeded"
```
Comment (ryanzhang-oss) [inline pkg/controllers/apply_controller.go:68]: those don't need to be const if it's not referenced by multiple places and needs to stay the same for a reasonable amount of time.

**Review by ryanzhang-oss — COMMENTED** [review]
please add klog level following fleet-network repo examples

**Comment by Ealianis** [comment]
Closing PR and splitting into different log PR and Event PR.

File: pkg/controllers/common.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package controllers
```
Comment (Ealianis) [inline pkg/controllers/common.go:1]: Moving to /utils.

**Pull request closed without merging**